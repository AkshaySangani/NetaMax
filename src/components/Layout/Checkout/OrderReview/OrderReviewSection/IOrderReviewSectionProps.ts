export interface IOrderReviewSectionProps {
  /**
   * Title
   * @type {string}
   */
  title: string;

  /**
   * Title icon
   * @type {ReactElement}
   */
  icon?: JSX.Element;
}
