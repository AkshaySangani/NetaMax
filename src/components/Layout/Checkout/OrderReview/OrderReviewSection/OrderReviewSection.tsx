import { Box, Flex, Text } from "@chakra-ui/react";
import { ReactElement } from "react";
import { IOrderReviewSectionProps } from "./IOrderReviewSectionProps";

/**
 * The OrderReviewSection component.
 * @param { IOrderReviewSectionProps } props the component props.
 * @returns { ReactElement } The OrderReviewSection component.
 */
export const OrderReviewSection: React.FC<IOrderReviewSectionProps> = (props): ReactElement => {
  const { title, children, icon } = props;

  return (
    <Box marginTop="40px">
      <Flex alignItems="center">
        <Text fontSize="32px" fontWeight="800" alignSelf="start">
          {title}
        </Text>
        {icon && icon}
      </Flex>
      {children}
    </Box>
  );
};
