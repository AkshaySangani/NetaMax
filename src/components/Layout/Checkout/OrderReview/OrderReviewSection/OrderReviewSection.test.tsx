import { cleanup, render, screen } from "@testing-library/react";
import { IOrderReviewSectionProps } from "./IOrderReviewSectionProps";
import { OrderReviewSection } from "./OrderReviewSection";

describe("OrderReviewSection test component", () => {
  let mock: IOrderReviewSectionProps;
  beforeEach(() => {
    mock = {
      title: "Section One",
    };
  });

  afterEach(() => {
    cleanup();
  });

  test("OrderReviewSection should be render correctly", () => {
    const { container } = render(<OrderReviewSection {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("OrderReviewSection title is on page", () => {
    render(<OrderReviewSection {...mock} />);
    expect(screen.getByText("Section One")).toBeInTheDocument();
  });
});
