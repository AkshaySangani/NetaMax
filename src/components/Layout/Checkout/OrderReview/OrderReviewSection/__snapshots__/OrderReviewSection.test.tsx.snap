// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderReviewSection test component OrderReviewSection should be render correctly 1`] = `
<div>
  <div
    class="css-y1gt6f"
  >
    <div
      class="css-70qvj9"
    >
      <p
        class="chakra-text css-1tsq2ie"
      >
        Section One
      </p>
    </div>
  </div>
</div>
`;
